modules/claims_api/app/controllers/claims_api/v1/forms/intent_to_file_controller.rb

```diff
@@ -46,17 +46,18 @@ def submit_form_0966
         # @return [JSON] Response from BGS
         def active
           check_for_type
-
-          bgs_response = bgs_service.intent_to_file.find_intent_to_file_by_ptcpnt_id_itf_type_cd(
+          message = "No Intent to file is on record for #{target_veteran_name} of type #{active_param}"
+          bgs_response = local_bgs_service.find_intent_to_file_by_ptcpnt_id_itf_type_cd(
             target_veteran.participant_id,
             ClaimsApi::IntentToFile::ITF_TYPES_TO_BGS_TYPES[active_param]
           )
+          raise ::Common::Exceptions::ResourceNotFound.new(detail: message) if bgs_response.blank?
+
           bgs_active = if bgs_response.is_a?(Array)
                          bgs_response.detect { |itf| active?(itf) }
                        elsif active?(bgs_response)
                          bgs_response
                        end
```

Comment:
I'm curious if we need this anymore as well. The new logic would never return anything besides an array.